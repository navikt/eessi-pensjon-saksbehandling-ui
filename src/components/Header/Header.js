import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { Icons, Nav } from 'eessi-pensjon-ui'
import * as routes from 'constants/routes'
import NavLogoTransparent from 'resources/images/NavLogoTransparent'
import './Header.css'

const Header = ({ actions, gettingUserInfo, header, history, isLoggingOut, t, username }) => {
  const onLogoClick = () => {
    actions.clearData()
    history.push({
      pathname: routes.ROOT,
      search: window.location.search
    })
  }

  const handleUsernameSelectRequest = (e) => {
    if (e.target.value === 'logout') {
      actions.clearData()
      actions.logout()
    }
  }

  return (
    <>
      <header className='c-topHeader'>
        <div className='brand'>
          <a href='#index' id='c-topHeader__logo-link' onClick={onLogoClick}>
            <NavLogoTransparent width='100' height='45' color='white' />
          </a>
          <div className='skillelinje' />
          <div className='tittel'><span>{t('app-headerTitle')}</span></div>
        </div>
        <div className='user'>
          {isLoggingOut
            ? <Nav.Spinner type='XS' />
            : (
              <div className={classNames('mr-2', 'SAKSBEHANDLER')}>
                <Icons kind='user' />
              </div>
            )}
          <div className='skillelinje' />
          <div className='mr-4 ml-2 align-middle name'>
            {gettingUserInfo ? t('buc:loading-gettingUserInfo')
              : username
                ? (
                  <Nav.Select
                    id='username-select-id'
                    className='username-select'
                    label=''
                    value={username}
                    selected={username}
                    onChange={handleUsernameSelectRequest}
                  >
                    <option value=''>{username}</option>
                    <option value='feedback'>{t('ui:giveFeedback')}</option>
                    <option value='logout'>{t('logout')}</option>
                  </Nav.Select>
                )
                : (
                  <>
                    <Icons kind='advarsel' size={16} />
                    <span className='username-span'>{t('unknown')}</span>
                  </>
                )}
          </div>
        </div>
      </header>
      {header
        ? (
          <Nav.Systemtittel className='m-4'>
            {header}
          </Nav.Systemtittel>
        )
        : null}
    </>
  )
}

Header.propTypes = {
  actions: PT.object.isRequired,
  gettingUserInfo: PT.bool,
  header: PT.oneOfType([PT.node, PT.string]),
  history: PT.object,
  isLoggingOut: PT.bool,
  t: PT.func.isRequired,
  username: PT.string
}

export default Header
