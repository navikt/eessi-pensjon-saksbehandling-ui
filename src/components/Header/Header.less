@import '../../constants';
@import (reference) '~nav-frontend-core/less/_variabler';

.c-topHeader {

    background-color: @navMorkGra;
    display: flex;
    flex-direction: row;
    height: 4rem;
    justify-content: space-between;
    flex-shrink: 0;

    .skillelinje {
      border-left: 1px solid black;
      display: flex;
      height: 30px;
      width: 1px;
    }

    .brand {
      align-items: center;
      display: flex;
      flex-direction: row;

      .logo {
        display: flex;
        height: auto;
        padding-left: 20px;
        padding-right: 15px;
        width: 85px;
      }

      .tittel {
        color: white;
        display: flex;
        font-size: 13pt;
        padding-left: 15px;
      }
    }

    .user {
      align-items: flex-end;
      display: flex;
      flex-direction: row;
      align-items: center;

      .SAKSBEHANDLER {
         color: red;
      }

      .BRUKER {
         color: green;
      }

      .username-span {
        padding: 0.45rem;
        padding-left: 0.5rem;
        padding-right: 1.6rem;
        color: white;
      }

      .username-select {

        .selectContainer:before,
        .selectContainer:after {
            background: white !important;
        }

        position: relative;
        select {
           background: transparent !important;

           &:not(:hover) {
             border-color: transparent !important;
           }
           &:hover {
             border-color: white !important;
           }
        }
        color: white;

        option {
            background: @pension-orange;
            padding: 0.5rem;
        }
      }

      .name {
        margin: auto 0px;
        padding: 0.3em;
      }
    }

  .skjemaelement__input {
    color: white;
  }
}

.highContrast .c-topHeader {
   background-color: @black;

   .skillelinje {
      border-left: 1px solid white;
   }
}
