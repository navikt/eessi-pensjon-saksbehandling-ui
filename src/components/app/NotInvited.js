import React, { Component } from 'react'
import { connect } from 'react-redux'
import PT from 'prop-types'
import { withNamespaces } from 'react-i18next'
import classNames from 'classnames'

import TopContainer from '../../components/ui/TopContainer/TopContainer'
import FrontPageDrawer from '../../components/drawer/FrontPage'

const mapStateToProps = (state) => {
  return {
    status: state.status
  }
}

class NotInvited extends Component {
  render () {
    const { t, history, location, status } = this.props

    return <TopContainer className={classNames('p-app-notInvited')}
      history={history} location={location}
      sideContent={<FrontPageDrawer t={t} status={status} />}>
      <h1 className='typo-sidetittel ml-0 appTitle'>{'not invited'}</h1>
    </TopContainer>
  }
}

NotInvited.propTypes = {
  t: PT.func.isRequired,
  history: PT.object,
  location: PT.object,
  status: PT.object
}

export default connect(
  mapStateToProps
)(
  withNamespaces()(NotInvited)
)
