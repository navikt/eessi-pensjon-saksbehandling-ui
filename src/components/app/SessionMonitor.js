import React, { Component } from 'react'
import PT from 'prop-types'
import { connect, bindActionCreators } from 'store'
import { withTranslation } from 'react-i18next'

import * as uiActions from '../../actions/ui'

const mapStateToProps = (state) => {
  return {
    loggedTime: state.app.loggedTime,
    expirationTime: state.app.expirationTime
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, uiActions), dispatch) }
}

export class SessionMonitor extends Component {
  componentDidMount () {
    this.checkTimeout()
  }

  closeModal () {
    const { actions } = this.props
    actions.closeModal()
  }

  checkTimeout () {
    let self = this
    const { t, actions, loggedTime, expirationTime, sessionExpiringWarning, sessionExpiredReload, checkInterval } = this.props
    if (!loggedTime || !sessionExpiringWarning || !sessionExpiredReload || !checkInterval) {
      return
    }
    return setTimeout(() => {
      const now = new Date()
      const diff = now.getTime() - loggedTime.getTime()
      if (diff > sessionExpiredReload) {
        window.location.reload()
      } else {
        if (diff > sessionExpiringWarning) {
          const minutesRemaining = Math.abs(Math.ceil((expirationTime.getTime() - now.getTime()) / 1000 / 60))
          actions.openModal({
            modalTitle: t('ui:session-expire-title'),
            modalText: t('ui:session-expire-text', { minutes: minutesRemaining }),
            modalButtons: [{
              main: true,
              text: t('ui:ok-got-it'),
              onClick: this.closeModal.bind(this)
            }]
          })
        }
        self.checkTimeout()
      }
    }, checkInterval)
  }

  render () {
    return <div />
  }
}

SessionMonitor.propTypes = {
  loggedTime: PT.object,
  expirationTime: PT.object,
  sessionExpiringWarning: PT.number,
  sessionExpiredReload: PT.number,
  checkInterval: PT.number
}

const ConnectedSessionMonitor = connect(
  mapStateToProps,
  mapDispatchToProps
)(
  withTranslation()(SessionMonitor)
)

export default ConnectedSessionMonitor
