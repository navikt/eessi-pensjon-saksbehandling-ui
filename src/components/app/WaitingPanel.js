import React, { Component } from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { withNamespaces } from 'react-i18next'
import * as Nav from '../ui/Nav'

class WaitingPanel extends Component {
  render () {
    const { t, message } = this.props

    return <div className={classNames('text-center')}>
      <Nav.NavFrontendSpinner />
      <p className='typo-normal'>{t(message)}</p>
    </div>
  }
}

WaitingPanel.propTypes = {
  message: PT.string.isRequired
}

export default withNamespaces(WaitingPanel)
