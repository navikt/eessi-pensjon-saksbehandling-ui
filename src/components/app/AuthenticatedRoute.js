import React, { Component } from 'react'
import PT from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Route, withRouter, Redirect } from 'react-router'
import _ from 'lodash'

import { IS_DEVELOPMENT } from '../../constants/environment'
import WaitingPanel from './WaitingPanel'

import * as routes from '../../constants/routes'
import * as appActions from '../../actions/app'
import * as statusActions from '../../actions/status'

const mapStateToProps = (state) => {
  return {
    userRole: state.app.userRole,
    userStatus: state.app.userStatus,
    loggedIn: state.app.loggedIn,
    allowed: state.app.allowed,
    isLoggingIn: state.loading.isLoggingIn,
    gettingUserInfo: state.loading.gettingUserInfo,
    rinaId: state.status.rinaId
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, appActions, statusActions), dispatch) }
}

const paramAliases = {
  'rinaid': 'rinaId',
  'saksNr': 'saksId',
  'fnr': 'aktoerId'
}

class AuthenticatedRoute extends Component {
  state = {
    isReady: false
  }

  parseSearchParams () {
    const { actions, location } = this.props

    let params = new URLSearchParams(location.search)
    let newParams = {}
    params.forEach((value, key) => {
      const _key = paramAliases.hasOwnProperty(key) ? paramAliases[key] : key
      const _value = value || undefined
      if (_value !== this.state[_key]) {
        actions.setStatusParam(_key, _value)
        newParams[_key] = _value
      }
    })
    if (!_.isEmpty(newParams)) {
      this.setState(newParams)
    }
    return newParams
  }

  componentDidMount () {
    const { actions, userStatus } = this.props
    if (!userStatus) {
      actions.getUserInfo()
    } else {
      this.setState({
        isReady: true
      })
    }
    this.parseSearchParams()
  }

  componentDidUpdate () {
    const { userStatus } = this.props
    const { isReady } = this.state

    if (!isReady && userStatus !== undefined) {
      this.setState({
        isReady: true
      })
    }
    this.parseSearchParams()
  }

  hasApprovedRole () {
    const { roles, userRole } = this.props
    return roles.indexOf(userRole) >= 0
  }

  comesFromPesys () {
    return this.state.hasOwnProperty('saksId') && this.state.hasOwnProperty('aktoerId')
  }

  forceLogin () {
    const { actions } = this.props

    console.log("No oidc-token, force login.")
    actions.login()
  }

  render () {
    const { userRole, allowed, gettingUserInfo } = this.props
    const { isReady } = this.state

    if (!isReady || gettingUserInfo) {
      return <WaitingPanel message='authenticating' />
    }

    if (!userRole) {
      this.forceLogin()
      return null
    }

    let validRole = this.hasApprovedRole()

    if (!validRole) {
      return <Redirect to={routes.FORBIDDEN} />
    }

    let authorized = allowed || IS_DEVELOPMENT

    if (!authorized) {
      return <Redirect to={routes.NOT_INVITED} />
    }

    return <Route {...this.props} />
  }
}

AuthenticatedRoute.propTypes = {
  className: PT.string,
  actions: PT.object.isRequired,
  roles: PT.array.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(
  withRouter(AuthenticatedRoute)
)
