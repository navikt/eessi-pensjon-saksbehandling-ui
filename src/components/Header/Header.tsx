import { clearData, logout } from 'actions/app'
import { toggleHighContrast, toggleSnow } from 'actions/ui'
import classNames from 'classnames'
import * as routes from 'constants/routes'
import Ui from 'eessi-pensjon-ui'
import PT from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import NavLogoTransparent from 'resources/images/NavLogoTransparent'
import './Header.css'

export interface HeaderProps {
  className ?: string;
  children?: JSX.Element | Array<JSX.Element | null>;
  gettingUserInfo?: boolean;
  header?: JSX.Element;
  isLoggingOut?: boolean;
  snow?: boolean;
  username?: string;
}

const Header: React.FC<HeaderProps> = ({
  className, children, gettingUserInfo, header,
  isLoggingOut, snow, username
}: HeaderProps): JSX.Element => {
  const dispatch = useDispatch()
  const { t } = useTranslation()
  const history = useHistory()

  const onLogoClick = () => {
    dispatch(clearData())
    history.push({
      pathname: routes.ROOT,
      search: window.location.search
    })
  }

  const onHighContrastClick = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault()
    e.stopPropagation()
    dispatch(toggleHighContrast())
  }

  const onSnowClick = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault()
    e.stopPropagation()
    dispatch(toggleSnow())
  }

  const handleUsernameSelectRequest = (e: React.ChangeEvent<HTMLSelectElement>) => {
    if (e.target.value === 'logout') {
      dispatch(clearData())
      dispatch(logout())
    }
  }

  return (
    <header role='banner'>
      <div className={classNames(className, 'c-topHeader')}>
        <div className='brand'>
          <a href='#index' id='c-topHeader__logo-link' onClick={onLogoClick}>
            <NavLogoTransparent width='100' height='45' color='white' />
          </a>
          <div className='skillelinje' />
          <div className='tittel'><span>{t('app-headerTitle')}</span></div>
        </div>
        <div>
          <Ui.Nav.Lenke
            id='c-topHeader__highcontrast-link-id'
            className='c-topHeader__highcontrast-link c-topHeader__link mt-1'
            href='#highContrast'
            onClick={onHighContrastClick}
          >
            {t('ui:highContrast')}
          </Ui.Nav.Lenke>
          <Ui.Nav.Lenke
            id='c-topHeader__snow-link-id'
            className='c-topHeader__snow-link c-topHeader__link ml-4 mt-1'
            href='#snow'
            onClick={onSnowClick}
          >
            {!snow ? 'la det snø!' : 'nok med snø!'}
          </Ui.Nav.Lenke>
        </div>
        <div className='user'>
          {isLoggingOut
            ? <Ui.Nav.Spinner type='XS' />
            : (
              <div className={classNames('mr-2', 'SAKSBEHANDLER')}>
                <Ui.Icons kind='user' />
              </div>
            )}
          <div className='skillelinje' />
          <div className='mr-4 ml-2 align-middle name'>
            {gettingUserInfo ? t('buc:loading-gettingUserInfo')
              : username
                ? (
                  <Ui.Nav.Select
                    id='username-select-id'
                    className='username-select'
                    label=''
                    value={username}
                    selected={username}
                    onChange={handleUsernameSelectRequest}
                  >
                    <option value=''>{username}</option>
                    <option value='feedback'>{t('ui:giveFeedback')}</option>
                    <option value='logout'>{t('logout')}</option>
                  </Ui.Nav.Select>
                )
                : (
                  <>
                    <Ui.Icons kind='advarsel' size={16} />
                    <span className='username-span'>{t('unknown')}</span>
                  </>
                )}
          </div>
        </div>
      </div>
      {header
        ? (
          <Ui.Nav.Systemtittel className='m-4'>
            {header}
          </Ui.Nav.Systemtittel>
        )
        : null}
      {children}
    </header>
  )
}

Header.propTypes = {
  className: PT.string,
  gettingUserInfo: PT.bool,
  header: PT.element,
  isLoggingOut: PT.bool,
  username: PT.string
}

export default Header
