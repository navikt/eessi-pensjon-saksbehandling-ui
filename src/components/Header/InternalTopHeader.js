import React, { Component } from 'react'
import { connect, bindActionCreators } from 'store'
import PT from 'prop-types'
import classNames from 'classnames'

import Icons from 'components/Icons'
import { NavFrontendSpinner, Select, Systemtittel } from 'components/Nav'
import * as routes from 'constants/routes'
import AdvarselTrekant from 'resources/images/AdvarselTrekant'
import NavLogoTransparent from 'resources/images/NavLogoTransparent'
import * as appActions from 'actions/app'
import * as uiActions from 'actions/ui'
import { getDisplayName } from 'utils/displayName'

import './InternalTopHeader.css'

const mapStateToProps = (state) => {
  return {
    username: state.app.username,
    highContrast: state.ui.highContrast,
    gettingUserInfo: state.loading.gettingUserInfo,
    isLoggingOut: state.loading.isLoggingOut
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, uiActions, appActions), dispatch) }
}

export class InternalTopHeader extends Component {
  onLogoClick () {
    const { history, actions } = this.props
    actions.clearData()
    history.push({
      pathname: routes.INDEX,
      search: window.location.search
    })
  }

  handleUsernameSelectRequest (e) {
    const { actions } = this.props

    if (e.target.value === 'logout') {
      actions.clearData()
      actions.logout()
    }
  }

  render () {
    let { t, username, gettingUserInfo, isLoggingOut, header } = this.props

    return <React.Fragment>
      <header className='c-ui-topHeader'>
        <div className='brand'>
          <a href='#index' id='c-ui-logo-link' onClick={this.onLogoClick.bind(this)}>
            <NavLogoTransparent width='100' height='45' color='white' />
          </a>
          <div className='skillelinje' />
          <div className='tittel'><span>{t('app-headerTitle')}</span></div>
        </div>
        <div className='user'>
          {isLoggingOut
            ? <NavFrontendSpinner type='XS' />
            : <div className={classNames('mr-2', 'SAKSBEHANDLER')}>
              <Icons kind='user' />
            </div>}
          <div className='skillelinje' />
          <div className='mr-4 ml-2 align-middle name'>
            {gettingUserInfo ? t('buc:loading-gettingUserInfo')
              : username
                ? <Select className='username-select'
                  label={''} value={username} selected={username}
                  onChange={this.handleUsernameSelectRequest.bind(this)}>
                  <option value=''>{username}</option>
                  <option value='feedback'>{t('ui:giveFeedback')}</option>
                  <option value='logout'>{t('logout')}</option>
                </Select>
                : <React.Fragment>
                  <AdvarselTrekant size={16} />
                  <span className='username-span'>{t('unknown')}</span>
                </React.Fragment>
            }
          </div>
        </div>
      </header>
      {header ? <Systemtittel className='m-4'>{header}</Systemtittel> : null}
    </React.Fragment>
  }
}

InternalTopHeader.propTypes = {
  t: PT.func.isRequired,
  username: PT.string,
  actions: PT.object,
  history: PT.object,
  gettingUserInfo: PT.bool,
  header: PT.oneOfType([PT.node, PT.string])
}

const ConnectedInternalTopHeader = connect(
  mapStateToProps,
  mapDispatchToProps
)(InternalTopHeader)

ConnectedInternalTopHeader.displayName = `Connect(${getDisplayName(InternalTopHeader)})`

export default ConnectedInternalTopHeader
