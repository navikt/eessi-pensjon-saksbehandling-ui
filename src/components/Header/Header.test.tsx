import { clearData } from 'src/actions/app'
import * as routes from 'src/constants/routes'
import { screen, render } from '@testing-library/react'
import { stageSelector } from 'src/setupTests'
import Header, { HeaderProps } from './Header'

jest.mock('actions/app', () => ({
  clearData: jest.fn()
}))

const mockHistoryPush = jest.fn()

jest.mock("react-router-dom", () => ({
  useNavigate: () => mockHistoryPush,
}));

describe('components/Header/Header', () => {
  const initialMockProps: HeaderProps = {
    username: 'testUser'
  }

  beforeEach(() => {
    stageSelector({}, {})
    render(<Header {...initialMockProps} />)
  })


  it('Render: match snapshot', () => {
    const { container } = render(<Header {...initialMockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Handling: clicking logo is handled', () => {
    (clearData as jest.Mock).mockReset();
    (mockHistoryPush as jest.Mock).mockReset()
    screen.getByTestId('c-header--logo-link').click()
    expect(clearData).toHaveBeenCalled()
    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: routes.ROOT,
      search: window.location.search
    })
  })
})
