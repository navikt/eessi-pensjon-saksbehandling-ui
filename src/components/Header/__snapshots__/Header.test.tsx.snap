// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header/Header Render: match snapshot 1`] = `
.c15 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g31[id="c14"] {
  content: "c15,";
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 * {
  font-size: inherit;
  line-height: inherit;
}

ui-fss.g49[id="c2"] {
  content: "c3,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

ui-fss.g50[id="c0"] {
  content: "c1,";
}

.c9 {
  font-size: 1.5rem;
}

ui-fss.g51[id="c8"] {
  content: "c9,";
}

.c17 {
  margin: auto 0px;
  padding: 0.3em;
}

ui-fss.g52[id="c16"] {
  content: "c17,";
}

.c13 {
  color: white;
}

ui-fss.g53[id="c12"] {
  content: "c13,";
}

.c5 {
  border-left: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 30px;
  width: 1px;
  margin-left: 1rem;
}

ui-fss.g54[id="c4"] {
  content: "c5,";
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 13pt;
  padding-left: 15px;
}

ui-fss.g55[id="c6"] {
  content: "c7,";
}

.c11 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

ui-fss.g56[id="c10"] {
  content: "c11,";
}

.c19 {
  padding: 0.45rem;
  padding-left: 0.5rem;
  padding-right: 1.6rem;
}

ui-fss.g57[id="c18"] {
  content: "c19,";
}

<Header
  highContrast={false}
  username="testUser"
>
  <styled.header
    role="banner"
  >
    <header
      className="c0 c1"
      role="banner"
    >
      <styled.div>
        <div
          className="c2 c3"
        >
          <a
            data-testid="c-header__logo-link"
            href="#index"
            onClick={[Function]}
          >
            <NavLogoTransparent
              color="white"
              height="45"
              width="100"
            >
              <svg
                height="45"
                version="1.1"
                viewBox="0 0 269 169"
                width="100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  NAV logo
                </title>
                <desc>
                  Created with Sketch.
                </desc>
                <defs>
                  <polygon
                    id="path-1"
                    points="22.407 43.4168 22.407 0.6878 0.5635 0.6878 0.5635 43.4168 22.407 43.4168"
                  />
                </defs>
                <g
                  fill="none"
                  fillRule="evenodd"
                  id="Modul-forslag"
                  stroke="none"
                  strokeWidth="1"
                >
                  <g
                    id="NAV-logo"
                  >
                    <g
                      id="Page-1-Copy"
                    >
                      <polygon
                        fill="white"
                        id="Fill-3"
                        points="0 121.3588 17.265 78.6298 33.854 78.6298 16.611 121.3588"
                      />
                      <polygon
                        fill="white"
                        id="Fill-5"
                        points="213.044 121.3588 230.088 78.6298 239.132 78.6298 222.089 121.3588"
                      />
                      <g
                        id="Group-9"
                        transform="translate(246.000000, 77.942000)"
                      >
                        <mask
                          fill="white"
                          id="mask-2"
                        >
                          <use
                            xlinkHref="#path-1"
                          />
                        </mask>
                        <g
                          id="Clip-8"
                        />
                        <polygon
                          fill="white"
                          id="Fill-7"
                          mask="url(#mask-2)"
                          points="0.5635 43.4168 17.6045 0.6878 22.4075 0.6878 5.3645 43.4168"
                        />
                      </g>
                      <path
                        d="M125.3091,168.942 C78.6681,168.942 40.8491,131.125 40.8491,84.477 C40.8491,37.824 78.6681,0 125.3091,0 C171.9671,0 209.7901,37.824 209.7901,84.477 C209.7901,131.125 171.9671,168.942 125.3091,168.942 Z M197.3604,78.6298 L182.3444,78.6298 C182.3444,78.6298 181.3094,78.6298 180.9434,79.5438 L172.6334,104.9828 L164.3304,79.5438 C163.9644,78.6298 162.9234,78.6298 162.9234,78.6298 L134.0514,78.6298 C133.4264,78.6298 132.9024,79.1518 132.9024,79.7728 L132.9024,88.4118 C132.9024,81.5588 125.6104,78.6298 121.3404,78.6298 C111.7784,78.6298 105.3774,84.9278 103.3844,94.5028 C103.2764,88.1508 102.7484,85.8748 101.0374,83.5438 C100.2514,82.4018 99.1154,81.4418 97.8784,80.6478 C95.3314,79.1558 93.0444,78.6298 88.1294,78.6298 L82.3584,78.6298 C82.3584,78.6298 81.3154,78.6298 80.9474,79.5438 L75.6964,92.5568 L75.6964,79.7728 C75.6964,79.1518 75.1764,78.6298 74.5524,78.6298 L61.1984,78.6298 C61.1984,78.6298 60.1674,78.6298 59.7924,79.5438 L54.3334,93.0748 C54.3334,93.0748 53.7884,94.4278 55.0344,94.4278 L60.1674,94.4278 L60.1674,120.2118 C60.1674,120.8518 60.6714,121.3588 61.3144,121.3588 L74.5524,121.3588 C75.1764,121.3588 75.6964,120.8518 75.6964,120.2118 L75.6964,94.4278 L80.8564,94.4278 C83.8174,94.4278 84.4444,94.5088 85.5964,95.0458 C86.2904,95.3078 86.9154,95.8378 87.2564,96.4488 C87.9544,97.7628 88.1294,99.3408 88.1294,103.9938 L88.1294,120.2118 C88.1294,120.8518 88.6434,121.3588 89.2784,121.3588 L101.9664,121.3588 C101.9664,121.3588 103.4004,121.3588 103.9674,119.9428 L106.7794,112.9928 C110.5184,118.2298 116.6724,121.3588 124.3204,121.3588 L125.9914,121.3588 C125.9914,121.3588 127.4344,121.3588 128.0054,119.9428 L132.9024,107.8148 L132.9024,120.2118 C132.9024,120.8518 133.4264,121.3588 134.0514,121.3588 L147.0034,121.3588 C147.0034,121.3588 148.4324,121.3588 149.0064,119.9428 C149.0064,119.9428 154.1864,107.0818 154.2064,106.9848 L154.2144,106.9848 C154.4134,105.9148 153.0614,105.9148 153.0614,105.9148 L148.4384,105.9148 L148.4384,83.8468 L162.9834,119.9428 C163.5514,121.3588 164.9834,121.3588 164.9834,121.3588 L180.2844,121.3588 C180.2844,121.3588 181.7244,121.3588 182.2924,119.9428 L198.4174,80.0138 C198.9754,78.6298 197.3604,78.6298 197.3604,78.6298 Z M132.9024,105.9148 L124.2024,105.9148 C120.7394,105.9148 117.9224,103.1108 117.9224,99.6438 C117.9224,96.1828 120.7394,93.3608 124.2024,93.3608 L126.6354,93.3608 C130.0894,93.3608 132.9024,96.1828 132.9024,99.6438 L132.9024,105.9148 Z"
                        fill="white"
                        id="Combined-Shape"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </NavLogoTransparent>
          </a>
          <styled.div>
            <div
              className="c4 c5"
            />
          </styled.div>
          <styled.div>
            <div
              className="c6 c7"
            >
              <span>
                ui:app-headerTitle
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
      <Styled(Lenke)
        data-testid="c-header__highcontrast-link-id"
        href="#highContrast"
        onClick={[Function]}
      >
        <Lenke
          className="c8 c9"
          data-testid="c-header__highcontrast-link-id"
          href="#highContrast"
          onClick={[Function]}
        >
          <a
            className="lenke c8 c9"
            data-testid="c-header__highcontrast-link-id"
            href="#highContrast"
            onClick={[Function]}
          >
            ui:highContrast
          </a>
        </Lenke>
      </Styled(Lenke)>
      <styled.div>
        <div
          className="c10 c11"
        >
          <styled.div>
            <div
              className="c12 c13"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f007",
                      "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                    ],
                    "iconName": "user",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user fa-w-14 "
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c14 c15"
            />
          </styled.div>
          <styled.div>
            <div
              className="c4 c5"
            />
          </styled.div>
          <styled.div>
            <div
              className="c16 c17"
            >
              <styled.span>
                <span
                  className="c18 c19"
                >
                  testUser
                </span>
              </styled.span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </header>
  </styled.header>
</Header>
`;
